core/sub.mk

```diff
@@ -50,3 +50,20 @@ recipe-embedded_secure_dtb = scripts/bin_to_c.py \
 				--out $(core-embed-fdt-c)
 $(eval $(call gen-dtb-file,$(core-embed-fdt-dts),$(core-embed-fdt-dtb)))
 endif
+
+ifeq ($(CFG_SHOW_CONF_ON_BOOT),y)
+conf-mk-xz-base64 := $(sub-dir-out)/conf.mk.xz.base64
+cleanfiles += $(conf-mk-xz-base64)
+
+$(conf-mk-xz-base64): $(conf-mk-file)
+	@$(cmd-echo-silent) '  GEN     $@'
+	$(q)tail +3 $< | xz | base64 -w 100 >$@ || rm $@
```

Comment:
For errors to propagate properly I think you need:
`s/rm $@/(rm -f $@ && false)/`

Reply:
Right. Using redirections in recipes is not so simple ;-)
I'm wondering, perhaps we should use `.DELETE_ON_ERROR` instead. Simpler and more robust.

Reply:
> perhaps we should use `.DELETE_ON_ERROR` instead.

I was unaware of that target, seems both useful and like a good candidate for the problem that we're trying to solve here.

Reply:
I've pushed an additional patch for this.